client-templates/azure-repos/accept.json.sample

```diff
@@ -0,0 +1,222 @@
+      }
+    },
+    {
+      "//": "create hook",
+      "method": "POST",
+      "path": "/_apis/hooks/subscriptions",
+      "origin": "https://${AZURE_REPOS_HOST}/${AZUR_REPOS_ORG}",
+      "auth": {
+        "scheme": "basic",
+        "token": "${BROKER_CLIENT_VALIDATION_BASIC_AUTH}"
+      }
+    },
+    {
+      "//": "delete hook",
+      "method": "DELETE",
+      "path": "/_apis/hooks/subscriptions/:subscriptionId",
+      "origin": "https://${AZURE_REPOS_HOST}/${AZUR_REPOS_ORG}",
+      "auth": {
+        "scheme": "basic",
+        "token": "${BROKER_CLIENT_VALIDATION_BASIC_AUTH}"
+      }
+    },
+    {
+      "//": "get file content. restrict by file types",
+      "method": "GET",
+      "path": "/:owner/_apis/git/repositories/:repo/items",
+      "origin": "https://${AZURE_REPOS_HOST}/${AZUR_REPOS_ORG}",
+      "valid": [
+        {
+          "queryParam": "path",
+          "values": [
+            "**/package.json",
+            "**%2Fpackage.json",
+            "**/yarn.lock",
+            "**%2Fyarn.lock",
+            "**/package-lock.json",
+            "**%2Fpackage-lock.json",
+            "**/Gemfile",
+            "**%2FGemfile",
+            "**/Gemfile.lock",
+            "**%2FGemfile.lock",
+            "**/pom.xml",
+            "**%2Fpom.xml",
+            "**/*req*.txt",
+            "**%2F*req*.txt",
+            "**/requirements/*.txt",
+            "**%2Frequirements%2F*.txt",
+            "**/build.gradle",
+            "**%2Fbuild.gradle",
+            "**/build.sbt",
+            "**%2Fbuild.sbt",
+            "**/.snyk",
+            "**%2F.snyk",
+            "**/packages.config",
+            "**%2Fpackages.config",
+            "**/*.csproj",
+            "**%2F*.csproj",
+            "**/*.vbproj",
+            "**%2F*.vbproj",
+            "**/*.fsproj",
+            "**%2F*.fsproj",
+            "**/project.json",
+            "**%2Fproject.json",
+            "**/Gopkg.toml",
+            "**%2FGopkg.toml",
+            "**/Gopkg.lock",
+            "**%2FGopkg.lock",
+            "**/vendor.json",
+            "**%2Fvendor.json",
+            "**/composer.lock",
+            "**%2Fcomposer.lock",
+            "**/composer.json",
+            "**%2Fcomposer.json",
+            "**/project.assets.json",
+            "**%2Fproject.assets.json",
+            "**/Podfile",
+            "**%2FPodfile",
+            "**/Podfile.lock",
+            "**%2FPodfile.lock"
+          ]
+        }
+      ],
+      "auth": {
+        "scheme": "basic",
+        "token": "${BROKER_CLIENT_VALIDATION_BASIC_AUTH}"
+      }
+    },
+    {
+      "//": "get list of globs for given repository",
+      "method": "GET",
+      "path": "/:owner/_apis/git/repositories/:repo/items",
+      "origin": "https://${AZURE_REPOS_HOST}/${AZUR_REPOS_ORG}",
+      "valid": [
+        {
+          "queryParam": "recursionLevel",
+          "values": ["full"]
+        }
+      ],
```

Comment:
if we remove this section, is every query param valid? Don't see the point in restricting query params when no param is a security issue.

I mean, if Snyk requested a single level instead of full it's not a security issue, so the Broker rules shouldn't be concerned by that. 

It would also let Snyk change the recursion in the future without changes to Broker.

Reply:
This is interesting one. 

Since Azure using same endpoint for fetching globs (paths) and getting file content, we need somehow restrict access to user's files. 
If we remove `valid ` array here, it basically will mean that we can pass any set of query parameters. For example: 
`?recursionLevel=full&download=true` that means fetch all files in repo. To avoid this behaviour we need to pass granular set of query parameters, that will exclude combinations of `recursionLevel=full` and `download=true`